src/quo/components/navigation/page_nav/style.cljs

```diff
@@ -14,12 +15,13 @@
 
 (defn center-content-container
   [centered? center-opacity]
-  {:flex              1
-   :margin-horizontal 12
-   :flex-direction    :row
-   :align-items       :center
-   :justify-content   (if centered? :center :flex-start)
-   :opacity           center-opacity})
+  (reanimated/apply-animations-to-style
+   (when center-opacity {:opacity center-opacity})
```

Comment:
Curious about how this might behave if `center-opacity` is falsy.

Reply:
hi @ibrkhalil good question. It will work, we will just not have animations.

If you check
https://github.com/status-im/status-mobile/blob/6a88a34d340e43063c75082113b4f6846a98ac9d/src/js/worklets/core.js#L39
We are looping over animations map (when has keys) and then performing operations.

Reply:
Hey @ajayesivan  Looks good, I just wanted to add a comment.

The function `apply-animations-to-style` is using the `use-animated-style` hook, and hooks have some rules, such as being called at the beginning of the component.

> Don’t call Hooks inside loops, conditions, or nested functions. Instead, always use Hooks at the top level of your React function, ...

Both [old react docs](https://legacy.reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level)  and [new docs](https://react.dev/reference/rules/rules-of-hooks#only-call-hooks-at-the-top-level) share the same idea.

To fix this issue I can see two approaches:

1. Move this call to the top level of the parent component, compute it once, and just pass the returned animated styles to each child, I'm not completely sure, but it seems the same calculation is being done multiple times :thinking: if so, this will also slightly improve the performance.

2. Remove the hook usage and just use the vector syntax for [inline animation styles](https://docs.swmansion.com/react-native-reanimated/docs/fundamentals/glossary#animations-in-inline-styling).

besides that, the code looks fine, wdyt?